packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -153,38 +153,47 @@ class CallWithChatContext {
  */
 export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapter {
   private callAdapter: CallAdapter;
-  private chatAdapterPromise: Promise<ChatAdapter>;
-  /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */
   private chatAdapter: ChatAdapter | undefined;
   private context: CallWithChatContext;
+  private emitter: EventEmitter = new EventEmitter();
   private onChatStateChange: (newChatAdapterState: ChatAdapterState) => void;
   private onCallStateChange: (newChatAdapterState: CallAdapterState) => void;
 
-  constructor(callAdapter: CallAdapter, chatAdapterPromise: Promise<ChatAdapter>) {
+  constructor(callAdapter: CallAdapter, chatAdapter?: ChatAdapter) {
     this.bindPublicMethods();
     this.callAdapter = callAdapter;
     this.context = new CallWithChatContext(callWithChatAdapterStateFromBackingStates(callAdapter));
 
     const onChatStateChange = (newChatAdapterState: ChatAdapterState): void => {
       this.context.updateClientStateWithChatState(newChatAdapterState);
     };
-    this.onChatStateChange = onChatStateChange;
 
-    this.chatAdapterPromise = chatAdapterPromise;
-    this.chatAdapterPromise.then((chatAdapter) => {
-      chatAdapter.onStateChange(this.onChatStateChange);
-      /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */
-      this.chatAdapter = chatAdapter;
-      this.context.updateClientStateWithChatState(chatAdapter.getState());
-    });
+    this.onChatStateChange = onChatStateChange;
+    if (chatAdapter) {
+      this.updateChatAdapter(chatAdapter);
+    }
 
     const onCallStateChange = (newCallAdapterState: CallAdapterState): void => {
       this.context.updateClientStateWithCallState(newCallAdapterState);
     };
+
     this.callAdapter.onStateChange(onCallStateChange);
     this.onCallStateChange = onCallStateChange;
   }
 
+  public setChatAdapterPromise(chatAdapter: Promise<ChatAdapter>): void {
+    chatAdapter.then((adapter) => {
+      this.updateChatAdapter(adapter);
```

Comment:
Add a check here to ensure we don't call `this.updateChatAdapter(adapter);` if the adapter has been disposed